source/server/configuration_impl.cc

```diff
@@ -90,8 +90,21 @@ void MainImpl::initialize(const envoy::config::bootstrap::v3::Bootstrap& bootstr
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, miss_timeout, 200));
   watchdog_megamiss_timeout_ =
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, megamiss_timeout, 1000));
-  watchdog_kill_timeout_ =
-      std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, kill_timeout, 0));
+
+  uint64_t kill_timeout = PROTOBUF_GET_MS_OR_DEFAULT(watchdog, kill_timeout, 0);
+  const uint64_t max_skew = PROTOBUF_GET_MS_OR_DEFAULT(watchdog, max_kill_skew, 0);
+
+  // Adjust kill timeout if we have skew enabled.
+  if (kill_timeout > 0 && max_skew > 0) {
+
+    // Increments the kill timeout with a random value in (0, max_skew).
```

Comment:
The skew is in the closed interval [1 ms, max_skew in msec] or semi closed interval (0 ms, max_skew in msec].

I recommend changing ) to ]

Reply:
Good Catch. Done.